diff -ur original/Launcher.qml patched/Launcher.qml
--- original/opt/lipstick-pandora/qml/Launcher.qml	2014-04-11 18:27:26.095184366 +0200
+++ patched/opt/lipstick-pandora/qml/Launcher.qml	2014-04-11 19:09:14.563565300 +0200
@@ -15,7 +15,7 @@
     id: gridview
 
     property Item reorderItem
-    property int columns: 4
+    property int columns: 5
     property int initialCellWidth: (desktop.width - Theme.paddingLarge * 2) / columns
     property real firstRowOpacity: Math.min(Math.max((dashboard.contentY - dashboard.lockscreenAndStatusAreaHeight)/cellHeight, 0.0), 1.0)
     property bool removeApplicationMode
@@ -255,10 +255,14 @@
                 NumberAnimation { target: launcherItem; property: "opacity"; to: 1.0; duration: 75 }
             }
 
-            Image {
+            Item {
                 id:icon
                 anchors.horizontalCenter: parent.horizontalCenter
-                y: Math.round(index <= 3 ? (parent.height / 2) - (height / 2) : (parent.height / 2) - ((height + launcherText.height) / 2))
+                y: Math.round(index < 5 ? (parent.height / 2) - (height / 2) : (parent.height / 2) - ((height + launcherText.height) / 2))
+                width: Theme.iconSizeLauncher
+                height: Theme.iconSizeLauncher
+            Image {
+                anchors.centerIn: parent
                 source: {
                     var iconId = model.object.iconId
                     if (iconId.indexOf(':/') !== -1) {
@@ -269,14 +273,13 @@
                         return 'image://theme/' + iconId
                     }
                 }
-                sourceSize.width: Theme.iconSizeLauncher
-                sourceSize.height: Theme.iconSizeLauncher
-                width: Theme.iconSizeLauncher
-                height: Theme.iconSizeLauncher
+                sourceSize.width: Theme.iconSizeLauncher * 0.8
+                sourceSize.height: Theme.iconSizeLauncher * 0.8
                 layer.effect: pressEffectComponent
                 layer.enabled: down
                 opacity: model.object.isUpdating ? 0.2 : 1.0
             }
+            }
 
             BusyIndicator {
                 anchors.centerIn: icon
