--- original/usr/share/jolla-settings/pages/sounds/VolumeSlider.qml	2014-04-17 15:41:33.717726166 +0200
+++ patched/usr/share/jolla-settings/pages/sounds/VolumeSlider.qml	2014-04-17 15:41:09.436544607 +0200
@@ -13,13 +13,65 @@
     label: qsTrId("settings_sounds_la_volume")
     maximumValue: 100
     minimumValue: 1
-    value: profileControl.ringerVolume
+    // value: profileControl.ringerVolume
 
-    onValueChanged: profileControl.ringerVolume = value
+    // onValueChanged: profileControl.ringerVolume = value
     onPressAndHold: cancel()
+    
+    Connections {
+        id: systemSlider
+        target: null
+        onValueChanged: settingsProfileControl.ringerVolume = slider.value
+    }
+    
+    Connections {
+        id: lipstickSlider
+        target: null
+        onValueChanged: profileControl.setVolumeLevel("general", slider.value)
+    }
 
+    Connections {
+        id: systemProfileControl
+        target: null
+        onRingerVolumeChanged: slider.value = settingsProfileControl.ringerVolume
+    }
+    
+    Connections {
+        id: lipstickProfileControl
+        target: null
+        onVolumeLevelChanged: {
+            if (profileName == "general") {
+                slider.value = level
+            }
+        }
+    }
+    
     ProfileControl {
-        id: profileControl
-        onRingerVolumeChanged: slider.value = profileControl.ringerVolume
+        id: settingsProfileControl
+        
+        
+        Component.onCompleted: {
+            for (var property in settingsProfileControl) {
+                if (property == "ringerVolume") {
+                    slider.value = settingsProfileControl.ringerVolume
+                    systemSlider.target = slider
+                    systemProfileControl.target = settingsProfileControl
+                }
+            }
+                        
+            // We are probably in lipstick
+            if (systemSlider.target != slider) {
+                if (typeof profileControl !== "undefined") {
+                    slider.value = profileControl.volumeLevel("general")
+                    lipstickSlider.target = slider
+                    lipstickProfileControl.target = profileControl
+                } else {
+                    slider.visible = false
+                    slider.enabled = false
+                    slider.height = 0
+                }
+            }
+        }
+        // onRingerVolumeChanged: slider.value = profileControl.ringerVolume
     }
 }
