--- original/opt/lipstick-pandora/qml/Lockscreen.qml	2014-04-11 18:27:26.099184398 +0200
+++ patched/opt/lipstick-pandora/qml/Lockscreen.qml	2014-04-17 09:59:13.895632740 +0200
@@ -9,6 +9,7 @@
 import org.nemomobile.lipstick 0.1
 import Sailfish.Silica 1.0
 import com.jolla.lipstick 0.1
+import org.nemomobile.time 1.0
 import "scripts/desktop.js" as Desktop
 
 MouseArea {
@@ -75,6 +76,22 @@
             offTimer.restart()
         }
     }
+    
+    Label {
+        visible: !ongoingCall.enabled
+        anchors.bottom: clock.top
+        anchors.horizontalCenter: parent.horizontalCenter
+        text: {
+            var dateString = Format.formatDate(wallClock.time, Format.DateFull)
+            return dateString.charAt(0).toUpperCase() + dateString.substr(1)
+        }
+        
+        WallClock {
+            id: wallClock
+            enabled: lockscreen.onDisplay
+            updateFrequency: WallClock.Day
+        }
+    }
 
     Clock {
         id: clock
@@ -188,6 +205,7 @@
         Behavior on opacity { FadeAnimation {} }
     }
     OngoingCall {
+        id: ongoingCall
         anchors.bottom: clock.top
     }
 }
--- original/opt/lipstick-pandora/qml/main.qml	2014-04-11 18:27:26.105184447 +0200
+++ patched/opt/lipstick-pandora/qml/main.qml	2014-04-16 13:18:27.064747763 +0200
@@ -395,7 +395,7 @@
                           : qsTrId("lipstick-jolla-home-me-sounds_off")
                     onClicked: { profileControl.profile = profileControl.profile == "silent" ? "ambience" : "silent" }
                 }
-                MenuLabel {
+                /*MenuLabel {
                     id: date
                     text: {
                         var dateString = Format.formatDate(wallClock.time, Format.DateFull)
@@ -406,7 +406,7 @@
                         enabled: pullDownMenu.active
                         updateFrequency: WallClock.Day
                     }
-                }
+                }*/
 
                 onActiveChanged: if (active) LipstickSettings.lockscreenVisible = true
             }
