--- original/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml	2014-04-16 20:11:29.235787000 +0300
+++ patched/usr/share/lipstick-jolla-home-qt5/eventsview/EventsView.qml	2014-04-16 20:12:27.938427102 +0300
@@ -15,6 +15,8 @@
 import Sailfish.Accounts 1.0
 import "../scripts/desktop.js" as Desktop
 import "../notifications"
+import com.jolla.settings 1.0
+import org.nemomobile.systemsettings 1.0
 
 ApplicationWindow {
     id: eventFeedWindow
@@ -162,6 +164,15 @@
                     width: parent.width
                     height: Theme.paddingLarge
                 }
+
+                FavoritesZone {
+                    id: favorites
+                }
+
+                Item {
+                    width: parent.width
+                    height: Theme.paddingLarge
+                }
                 EventsViewPostStatusField {
                     id: postStatusField
                     manager: accountManager
