--- original/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2014-06-08 21:55:01.847978384 +0200
+++ patched/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2014-06-15 12:28:18.029087835 +0200
@@ -479,7 +479,6 @@
                 color: down ? Theme.highlightColor : Theme.primaryColor
                 font.pixelSize: Theme.fontSizeTiny
                 text: object.title
-                visible: !launcherEditMode || itemType === LauncherModel.Folder
             }

             Component {
@@ -501,8 +500,8 @@
             Component {
                 id: uninstallButtonComponent
                 UninstallButton {
-                    anchors.verticalCenter: launcherIcon.bottom
-                    anchors.horizontalCenter: parent.horizontalCenter
+                    anchors.verticalCenter: launcherIcon.top
+                    anchors.right: parent.right
                     enabled: launcherEditMode
                     opacity: enabled ? 1.0 : 0.0
                     visible: launcherEditMode && itemType == LauncherModel.ApCplication && appControl.isUninstallable(object.filePath) && !object.isUpdating
