--- original/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2014-06-08 21:55:01.847978384 +0200
+++ patched/usr/share/lipstick-jolla-home-qt5/launcher/LauncherGrid.qml	2014-06-08 22:14:11.587192377 +0200
@@ -17,7 +17,7 @@
 
     property Item reorderItem
     property Item movingItem
-    property int columns: 4
+    property int columns: 5
     property int initialCellWidth: (desktop.width - Theme.paddingLarge * 2) / columns
     property real firstRowOpacity: Math.min(Math.max((dashboard.contentY - dashboard.lockscreenAndStatusAreaHeight)/cellHeight, 0.0), 1.0)
     property bool removeApplicationMode
@@ -31,7 +31,7 @@
     width: cellWidth * columns
     anchors.horizontalCenter: parent.horizontalCenter
     // Increase cellWidth so that icon vertical edges are Theme.paddingLarge away from display edges
-    cellWidth: Math.round(initialCellWidth + (initialCellWidth - Theme.iconSizeLauncher) / (columns - 1))
+    cellWidth: Math.round(initialCellWidth + (initialCellWidth - Theme.iconSizeLauncher * 0.8) / (columns - 1))
     cellHeight: Math.round(Screen.height / 6)
 
     function showFolder(folder) {
@@ -366,10 +366,14 @@
                 id: launcherIcon
                 anchors {
                     centerIn: parent
-                    verticalCenterOffset: rootFolder && index <= 3  ? 0 : Math.round(-launcherText.height/2)
+                    verticalCenterOffset: rootFolder && index < gridview.columns  ? 0 : Math.round(-launcherText.height/2)
                 }
                 icon: model.object.iconId
                 pressed: down
+                width: Theme.iconSizeLauncher * 0.8
+                height: Theme.iconSizeLauncher * 0.8
+                sourceSize.width: Theme.iconSizeLauncher * 0.8
+                sourceSize.height: Theme.iconSizeLauncher * 0.8
                 opacity: isUpdating && folderItemCount == 0 ? 0.2 : 1.0
                 Text {
                     font.pixelSize: Theme.fontSizeExtraLarge
