--- original/usr/share/lipstick-jolla-home-qt5/qml/AmbiencePage.qml	2014-06-08 21:55:01.862978516 +0200
+++ patched/usr/share/lipstick-jolla-home-qt5/qml/AmbiencePage.qml	2014-06-08 22:45:58.007084002 +0200
@@ -9,6 +9,8 @@
 import Sailfish.Silica 1.0
 import Sailfish.Ambience 1.0
 import Sailfish.Gallery 1.0
+import com.jolla.settings 1.0
+import org.nemomobile.systemsettings 1.0
 import org.nemomobile.configuration 1.0
 import org.nemomobile.thumbnailer 1.0
 import org.nemomobile.lipstick 0.1
@@ -32,6 +34,10 @@
                 && (ambienceSwitcherOpen || (ambienceSwitcherEnabled && ambienceModel.count > 1))
                 && (PeekFilter.activeArea == PeekFilter.Left || PeekFilter.activeArea == PeekFilter.Right) && Lipstick.compositor.homeActive
                 && !launcher.openedChildFolder && !desktop.closeApplicationEnabled
+    readonly property bool _switcherLockEnabled: (!progressAnimation.running && !contentAnimation.running)
+                && (ambienceSwitcherOpen || (ambienceSwitcherEnabled && ambienceModel.count > 1))
+                && (PeekFilter.activeArea == PeekFilter.Left || PeekFilter.activeArea == PeekFilter.Right) && Lipstick.compositor.homeActive
+                && !launcher.openedChildFolder && !desktop.closeApplicationEnabled
 
     property string _currentHomeWallpaper: Ambience.homeWallpaper
 
@@ -42,7 +48,9 @@
     }
 
     PeekFilter.onGestureStarted: {
-        if (_switcherEnabled) {
+        if ((_switcherEnabled && PeekFilter.activeArea == PeekFilter.Left)
+            || (_switcherLockEnabled && PeekFilter.activeArea == PeekFilter.Right)
+            || ambienceSwitcherOpen) {
             anchorAmbienceItem()
             progressBinding.when = true
         }
@@ -120,6 +128,16 @@
             ambienceItem.anchors.right = undefined
             ambienceItem.anchors.left = contentItem.right
         }
+        
+        if (!ambienceSwitcherOpen) {
+            if (PeekFilter.activeArea == PeekFilter.Right) {
+                gridView.visible = false
+                widgets.visible = true
+            } else {
+                gridView.visible = true
+                widgets.visible = false
+            }
+        }
     }
 
     property real _progress
@@ -210,6 +228,17 @@
         anchors.left: contentItem.right
         // Scenegraph does not seem to cull yet, explicitly hide
         visible: contentItem.x != 0 || page.ambienceSwitcherOpen
+        
+        Item {
+            id: widgets
+            visible: false
+            anchors.fill: parent
+            
+            FavoritesZone {
+                id: favorites
+                anchors.top: parent.top
+            }
+        }
 
         ImageGridView {
             id: gridView
