--- /usr/share/lipstick-jolla-home-qt5/lockscreen/Lockscreen.qml	2014-07-18 01:14:25.682334258 +0400
+++ /usr/share/lipstick-jolla-home-qt5/lockscreen/Lockscreen.qml	2014-07-18 01:16:55.609656142 +0400
@@ -122,12 +122,13 @@
             topMargin: Theme.paddingLarge
         }
         width: parent.width
-        height: Math.max(cellularNetworkNameStatusIndicator.height, dateLabel.height)
+        height: cellularNetworkNameStatusIndicator.height + dateLabel.height
 
         CellularNetworkNameStatusIndicator {
             id: cellularNetworkNameStatusIndicator
             anchors.horizontalCenter: parent.horizontalCenter
-            opacity: 1 - dateLabel.opacity
+	    anchors.top: dateLabel.bottom 
+            opacity: dateLabel.opacity
         }
 
         Label {
@@ -147,10 +148,11 @@
                 enabled: lockScreen.onDisplay
                 updateFrequency: WallClock.Day
             }
-            Behavior on opacity {
+            opacity: 1
+            /*Behavior on opacity {
                 id: dateOpacityBehavior
                 FadeAnimation { }
-            }
+            }*/
         }
     }
 
@@ -283,12 +285,12 @@
     Timer {
         id: hideDateTimer
         interval: 3000
-        onTriggered: {
+        /*onTriggered: {
             var indicator = cellularNetworkNameStatusIndicator
             if (!lipstickSettings.lowPowerMode && (indicator.homeNetwork || indicator.visitorNetwork)) {
                 lockScreen.state = "showCellularNetworkName"
             }
-        }
+        }*/
     }
 
     states: [
