--- /usr/share/lipstick-jolla-home-qt5/qml/Lockscreen.qml	2014-07-16 01:57:46.242647008 +0400
+++ /usr/share/lipstick-jolla-home-qt5/qml/Lockscreen.qml	2014-07-16 02:08:42.693373405 +0400
@@ -126,7 +126,8 @@
         CellularNetworkNameStatusIndicator {
             id: cellularNetworkNameStatusIndicator
             anchors.horizontalCenter: parent.horizontalCenter
-            opacity: 1 - dateLabel.opacity
+            anchors.top: dateLabel.bottom
+	    opacity: dateLabel.opacity
         }
 
         Label {
@@ -146,10 +147,11 @@
                 enabled: lockScreen.onDisplay
                 updateFrequency: WallClock.Day
             }
-            Behavior on opacity {
+            /*Behavior on opacity {
                 id: dateOpacityBehavior
                 FadeAnimation { }
-            }
+            }*/
+	    opacity: 1
         }
     }
 
@@ -313,7 +315,7 @@
             name: "showCellularNetworkName"
             PropertyChanges {
                 target: dateLabel
-                opacity: 0
+                opacity: 1
             }
         }
     ]
